lib/engine/game/g_1862/step/track.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/track'
+
+module Engine
+  module Game
+    module G1862
+      module Step
+        class Track < Engine::Step::Track
+          def lay_tile_action(action, entity: nil, spender: nil)
+            tile_lay = get_tile_lay(action.entity)
+            if action.tile.label.to_s == 'N' && !(tile_lay && tile_lay[:upgrade])
+              raise GameError, "Cannot lay an 'N' tile now"
+            end
+
+            super
+          end
+
+          def upgraded_track(action)
+            # this also takes care of adding small stations, since that is never yellow to yellow
+            @round.upgraded_track = true if action.tile.color != :yellow || action.tile.label.to_s == 'N'
+          end
+
+          def update_tile_lists(tile, old_tile)
+            @game.tiles.delete(tile)
+            @game.tiles << old_tile unless old_tile.preprinted
+
+            return unless tile.cities.empty? && tile.color != old_tile.color && tile.color != :yellow
+
+            # if an upgrade without cities => remove/restore base tile
+            new_base_name = @game.base_tile_name(tile)
+            new_base_tile = @game.tiles.find { |t| t.name == new_base_name }
+            @game.tiles.delete(new_base_tile)
+            @game.base_tiles << new_base_tile
+
+            return unless old_tile.color != :yellow
+
+            old_base_name = @game.base_tile_name(old_tile)
+            old_base_tile = @game.base_tiles.find { |t| t.name == old_base_name }
+            @game.base_tiles.delete(old_base_tile)
+            @game.tiles << old_base_tile
+          end
+
+          def potential_tiles(entity, hex)
+            colors = @game.phase.tiles
+            if normal_available_hex(entity, hex)
+              @game.tiles
+                .select { |tile| colors.include?(tile.color) }
+                .uniq(&:name)
+                .select { |t| @game.upgrades_to?(hex.tile, t) }
```

Comment:
doesn't seem like you need to copy and paste this, only need to change the lambda

Reply:
OK.